app/cdap/components/GridTable/types.ts

```diff
@@ -48,3 +48,21 @@ export interface IHeaderNamesList {
 export interface IType {
   [key: string]: string | number;
 }
+
+interface IGeneralObjectRecord {
+  [key: string]: number | string;
+}
+
+interface IGeneral {
+  general?: IGeneralObjectRecord;
+  types?: IGeneralObjectRecord;
+}
+
+export interface IStatistics {
+  [key: string]: IGeneral;
+}
+
+export interface IAddTransformationItem {
+  option: string;
+  supportedDataType: string[];
```

Comment:
what are these supported data types? I'm guessing its a small list of supported data types and not something that can be set by the user. If so - please list the included data types here - your code below will be a lot simpler if you don't have to filter through this or do .includes everywhere.

Reply:
Supported datatype will always differ based on the option / transformation selected, this is setting dynamically based on option for e.g if uppercase selected then supportedDataType will be ['string'],  if add (math option) is selected then it will be ['int', 'float', 'double', ..] if it can be applicable on any type then ['all']